src/main/java/io/nats/client/KeyValue.java

```diff
@@ -0,0 +1,121 @@
+// Copyright 2021 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package io.nats.client;
+
+import io.nats.client.api.KvEntry;
+
+import java.io.IOException;
+
+/**
+ * Key Value Store Management context for creation and access to key value buckets.
+ */
+public interface KeyValue {
+
+    /**
+     * Get the byte[] value for a key
+     * THIS IS A BETA FEATURE AND SUBJECT TO CHANGE
+     * @param key the key
+     * @return the value or null if not found or deleted
+     * @throws IOException covers various communication issues with the NATS
+     *         server such as timeout or interruption
+     * @throws JetStreamApiException the request had an error related to the data
+     * @throws IllegalArgumentException the server is not JetStream enabled
+     */
+    byte[] getValue(String key) throws IOException, JetStreamApiException;
+
+    /**
+     * Get the string value for a key
+     * THIS IS A BETA FEATURE AND SUBJECT TO CHANGE
+     * @param key the key
+     * @return the value as UTF-8 string or null if not found or deleted
+     * @throws IOException covers various communication issues with the NATS
+     *         server such as timeout or interruption
+     * @throws JetStreamApiException the request had an error related to the data
+     * @throws IllegalArgumentException the server is not JetStream enabled
+     */
+    String getStringValue(String key) throws IOException, JetStreamApiException;
```

Comment:
Just a suggestion, wondering if the base KV should simply stick to entries - if you do that, it opens the opportunity for a generics version where the value in the generic KV can be anything a codec can do.

```typescript
export interface RoKV {
  get(k: string): Promise<Entry | null>;
  history(k: string): Promise<QueuedIterator<Entry>>;
  watch(opts?: { key?: string }): Promise<QueuedIterator<Entry>>;
  close(): Promise<void>;
  status(): Promise<KvStatus>;
  keys(): Promise<string[]>;
}

export interface KV extends RoKV {
  put(k: string, data: Uint8Array, opts?: Partial<PutOptions>): Promise<number>;
  delete(k: string): Promise<void>;
  purge(opts?: PurgeOpts): Promise<PurgeResponse>;
  destroy(): Promise<boolean>;
}
```

The generics version can simply delegate to the entry to provide the user value they want to see
```typescript
export interface EncodedRoKV<T> {
  get(k: string): Promise<EncodedEntry<T> | null>;
  history(k: string): Promise<QueuedIterator<EncodedEntry<T>>>;
  watch(opts?: { key?: string }): Promise<QueuedIterator<EncodedEntry<T>>>;
  close(): Promise<void>;
  status(): Promise<KvStatus>;
  keys(): Promise<string[]>;
}

export interface EncodedKV<T> extends EncodedRoKV<T> {
  put(k: string, data?: T, opts?: Partial<PutOptions>): Promise<number>;
  delete(k: string): Promise<void>;
  purge(opts?: PurgeOpts): Promise<PurgeResponse>;
  destroy(): Promise<boolean>;
}
```

Reply:
The basic entry, can provide the standard primitive wrappers if you'd like. My guess is in most cases the values will match a bucket, but I might be wrong. For NGS I have different buckets, and while I could put anything in one bucket, I have the data separated into different buckets that have data types that are expected.